aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -13,51 +13,52 @@ namespace {
 
 using namespace vec256;
 
-template <class scalar_t, class F>
-static void parallel_apply(Tensor& result, const Tensor& self, F f) {
-  internal::init_tbb_num_threads();
-
-static default_partitioner_type ap;
-
-  auto arr_out = result.data<scalar_t>();
-  auto arr_in = self.data<scalar_t>();
-  int64_t size = self.numel();
-  int64_t grain_size = 2048;
-  if (size < grain_size) {
-    map(f, arr_out, arr_in, size);
-  } else {
-    tbb::parallel_for(
-        tbb::blocked_range<int64_t>(0, size, grain_size),
-        [&](const tbb::blocked_range<int64_t>& r) {
-          map(f, arr_out + r.begin(), arr_in + r.begin(), r.end() - r.begin());
-        },
-        ap);
-  }
-}
-
 static void abs_kernel(Tensor& result, const Tensor& self) {
   AT_DISPATCH_ALL_TYPES(self.type(), "abs", [&] {
     parallel_apply<scalar_t>(
         result,
         self,
-        [](const Vec256<scalar_t>& x) { return x.abs(); });  });
+        [](scalar_t* out, scalar_t* in, int64_t size) {
+          map([](const Vec256<scalar_t>& x) { return x.abs(); },
+              out, in, size);
+        },
+        2048);
+  });
 }
 
 static void rsqrt_kernel(Tensor& result, const Tensor& self) {
   AT_DISPATCH_FLOATING_TYPES(self.type(), "rsqrt", [&] {
     parallel_apply<scalar_t>(
         result,
         self,
-        [](const Vec256<scalar_t>& x) { return Vec256<scalar_t>((scalar_t)(1)) / x.sqrt(); });  });
+        [](scalar_t* out, scalar_t* in, int64_t size) {
+          map(
+              [](const Vec256<scalar_t>& x) {
+                return Vec256<scalar_t>((scalar_t)(1)) / x.sqrt();
+              },
+              out,
+              in,
+              size);
+        },
+        2048);
```

Comment:
This used to be some `grain_size` constant, now it has changed. Why? Can you define it as a `constexpr` constant instead, and use it here?

Reply:
We didn't have enough parallelization for some functions. For now it's a constant throughout but soon enough it'll represent an estimate of operations per element to discount the default grainsize.